### PR TITLE
Auth commands

### PR DESCRIPTION
**Pull request by ppiques**

### CODE REVIEW

File: include/IRCErrors.hpp [hunk 1]
```diff
@@ -19,4 +19,43 @@ class ERR_UNKNOWNCOMMAND : public std::exception
...
+{
+	std::string _str;
+	ERR_NEEDMOREPARAMS(const std::string &command) : _str(command + " :Not enough parameters") {}
+	~ERR_NEEDMOREPARAMS(void) throw() {}
+	static const int code = 461;
+	virtual const char* what() const throw()
+	{
+		return this->_str.c_str();
+	}
+};
```
Comment (vfurmane) [inline include/IRCErrors.hpp:41]: This should be a class (`_str` would be private)

File: include/Message.hpp [hunk 2]
```diff
@@ -20,7 +20,12 @@ struct Message
 
 	void		parse();
 	void		execute();
-	
+
+	size_t				getArgsCount(void) const;
+	const std::string	*getArguments(void) const;
+	const std::string	&getInput(void) const;
+	const std::string	&updateInputFromFields(void);
+
```
Comment (vfurmane) [inline include/Message.hpp:28]: Getters are useless, this is a structure

**Review by theovgl — CHANGES_REQUESTED** [review]

File: include/Peer.hpp [hunk 3]
```diff
@@ -6,8 +6,11 @@
 # include <stdint.h>
 # include <string>
 # include <sys/socket.h>
+# include <sys/types.h>
```
Comment (vfurmane) [inline include/Peer.hpp:9]: Useless include

File: include/Peer.hpp [hunk 4]
```diff
@@ -21,6 +24,10 @@ class Peer
 		void				clearMessage(void);
 		bool				hasCompleteMessage(void) const;
 		const std::string	&getMessage(void) const;
+		bool				isRegistered(void) const;
+		void				registration(const std::string &user, const std::string &mode, const std::string &realname);
```
Comment (vfurmane) [inline include/Peer.hpp:28]: `register` is a better name according to me. What do you think @theovgl ?

File: include/commands.hpp [hunk 5]
```diff
@@ -1,6 +1,26 @@
...
+{
+	IRCPeerManager	&peers;
+};
+
+struct NickMessage : public IRCMessage
+{
+	NickMessage(IRCPeer &peer, const std::string &new_nick);
+};
+
+void	command_nick(IRCMessage &message, Dependencies &deps);
```
Comment (vfurmane) [inline include/commands.hpp:24]: Missing `command_user` and `command_quit`

File: src/Message.cpp [hunk 6]
```diff
@@ -89,3 +89,34 @@ void	Message::execute()
...
+
+const std::string	*Message::getArguments(void) const
+{
+	return this->arguments;
+}
+
+const std::string	&Message::getInput(void) const
+{
+	return this->input;
+}
```
Comment (vfurmane) [inline src/Message.cpp:106]: Those are useless getters

File: src/commands.cpp [hunk 7]
```diff
@@ -0,0 +1,16 @@
+#include "commands.hpp"
+#include "IRCMessage.hpp"
+#include "IRCPeer.hpp"
+
+ErrorMessage::ErrorMessage(IRCPeer &peer, const std::string &quit_message) : IRCMessage(peer, std::string())
```
Comment (vfurmane) [inline src/commands.cpp:5]: `quit_message` (the name) is irrelevant

File: src/commands.cpp [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+{
+	(void)quit_message;
+}
```
Comment (vfurmane) [inline src/commands.cpp:8]: Where is the implementation ?

File: test/TCP/Server.test.cpp [hunk 9]
```diff
@@ -0,0 +1,5 @@
+#include "catch.hpp"
+#include <cstdio>
+
+#define private public
+#include "TCP/Server.hpp"
```
Comment (vfurmane) [inline test/TCP/Server.test.cpp:5]: @theovgl

**Review by vfurmane — CHANGES_REQUESTED** [review]
The Makefile does not compile `commands/*.cpp` files.

The commands are not integrated into the executor.

File: include/Peer.hpp (hunk 4 above)
Comment (ppiques) [reply include/Peer.hpp:28]: c'est noté.

**Review by theovgl — APPROVED** [review]

File: src/Server.cpp [hunk 10]
```diff
@@ -66,7 +66,7 @@ void		Server::_bindNewSocketToPort(char *port)
 #endif
 			continue;
 		}
-#ifndef NDEBUG
+#ifndef NDEBUGregistration
```
Comment (vfurmane) [inline src/Server.cpp]: typo

**Review by vfurmane — CHANGES_REQUESTED** [review]
The commands are not integrated into the executor.

**Review by vfurmane — APPROVED** [review]

**Pull request merged**